net/tcp/tcp_send.c

```diff
@@ -329,6 +331,32 @@ static void tcp_sendcommon(FAR struct net_driver_s *dev,
     }
 #endif /* CONFIG_NET_IPv4 */
 
+  /* The initial conn->rcvseq and conn->sndseq values are defined to 0 and
+   * they are copied to tcp->ackno and tcp->seqno respectively. According
+   * to RCF1948 the initial TCP sequence number should be a random value and
+   * some servers will not accept TCP seqno starting from value 0.
+   *
+   * This fix solves this issue, however after a seqno/ackno overflow this
+   * value will be 0 again and in this case this packet will be discarded.
+   * This is not big issue because it only will happen after transmitting
+   * more than 2 billions packages (see tcp_initsequence).
+   *
+   * FIXME: This check should be done only at start of connection, but I
+   *        tried to put it at tcp_connect() but it didn't work.
+   */
+
+  tmpseq = tcp_getsequence(conn->rcvseq);
+  if (tmpseq == 0)
+    {
+      tcp_initsequence(conn->rcvseq);
+    }
```

Comment:
Why this is applied to `rcvseq` as well?

Reply:
to make ACK number random as well